packages/vx-xychart/src/components/series/LineSeries.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useContext, useCallback, useEffect } from 'react';
+import LinePath from '@vx/shape/lib/shapes/LinePath';
+import { ScaleConfig, ScaleConfigToD3Scale, ScaleInput } from '@vx/scale';
+import { AxisScaleOutput } from '@vx/axis';
+import DataContext from '../../context/DataContext';
+import isValidNumber from '../../typeguards/isValidNumber';
+
+type LineSeriesProps<
+  XScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  YScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  Datum
+> = {
+  dataKey: string;
+  data: Datum[];
+  xAccessor: (
+    d: Datum,
+  ) => // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ScaleInput<ScaleConfigToD3Scale<XScaleConfig, AxisScaleOutput, any, any>>;
+  yAccessor: (
+    d: Datum,
+  ) => // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ScaleInput<ScaleConfigToD3Scale<YScaleConfig, AxisScaleOutput, any, any>>;
+};
+
+export default function LineSeries<
+  XScaleConfig extends ScaleConfig<AxisScaleOutput>,
```

Comment:
One thing I am not sure here is how to pass generic from the `XYChart` to the `LineSeries` for stronger type-checking without manually doing

```
<LineSeries<A, B, C> data={} xAccessor={} yAccessor={} />
```

If I understand correctly, if generics are not passed like it is now, TS only tries to infer `XScaleConfig` and `YScaleConfig` back from `xAccessor` and `yAccessor`. Not sure if that backward inference is successful. It might give somewhat broad types.

Not blocking though. Worst-case is type is not 100% strict. Probably ok to revisit later.

Reply:
good question, I'm not sure of this either. I may play around with what the types end up being locally but can revisit when building more robust Series components (this was MVP)